### DISCUSSION TITLE
Producers and consumers of Metrics

### DISCUSSION THREAD

**Discussion by schnuerle** [openmobilityfoundation/curb-data-specification#29] [Data Specification]
Who will be collecting the event data needed and producing the Metrics using our defined [methodology](https://github.com/openmobilityfoundation/curb-data-specification/discussions/27)? We assume it is city agencies who are getting Event API data in real time or historically from sensors and sources. 

Who will be consuming the data coming out of Metrics? Likely it is other city/agency employees using the data to meet their curb program [use cases](https://github.com/openmobilityfoundation/curb-data-specification/discussions/28). 

Is it also commercial entities and other curb users? Who would be interested in aggregated curb metrics and counts at individual curb spaces/zones/areas? What would this data be used for?

**Comment by jacobmalleau** [comment]
Third party platforms that help city agencies visualize/ better understand these metrics will also be consuming this data. Provided that event data is publicly available and/or passed on to these platforms, they may also be the ones producing the metrics for cities. 

For example, at CurbIQ, we would adopt these APIs to utilize in our dashboards as well as help cities make use of their event data.

**Comment by bhamlinSDOT** [comment]
I anticipate city agencies would be collecting the event data, though it could be a designated third party at first while cities get their IT capabilities up to speed.

- Consumers will likely be cities or vendors who are processing the data for cities. 
  - If analyzing a particular zone I might want to know by time of day:
    - dwell time 
    - number of events per hour
    - % occupancy per 15 minutes

Having this data easily accessible via a dashboard would be amazing.

**Comment by schnuerle** [comment]
Given that Metrics now has [two endpoints](https://github.com/openmobilityfoundation/curb-data-specification/tree/feature-release-work-1/metrics#rest-endpoints), one with a subset of events and another with some aggregation, I think we can say that 1) a city/agency produces the endpoints and b) the consumers can be city employees, or others they give access to, including researchers, third parties, or even the public with open data.
